### PR TITLE
Optimize any_variants_not_track_inventory? method in product.rb

### PR DESCRIPTION
**Pull request by nishant-cyro**

### CODE REVIEW

File: core/app/models/spree/product.rb [hunk 1]
```diff
@@ -240,10 +240,11 @@ def add_associations_from_prototype
     end
 
     def any_variants_not_track_inventory?
+      return true unless Spree::Config.track_inventory_levels
       if variants_including_master.loaded?
         variants_including_master.any? { |v| !v.should_track_inventory? }
```
Comment (Mafi88) [inline core/app/models/spree/product.rb]: Just a thought... as we check ```Spree::Config.track_inventory_levels``` before this line anyway, wouldn't it be better to use here:
```
variants_including_master.any? { |v| !v.track_inventory? }
```

(to omit ```Spree::Config.track_inventory_levels``` check in https://github.com/spree/spree/blob/d831a86389e2c2e11abe2286a5ad4e8caa3252ff/core/app/models/spree/variant.rb#L217)

Comment (nishant-cyro) [inline core/app/models/spree/product.rb]: @Mafi88 - Done, can you please rebuild it?

**Comment by Mafi88** [comment]
@nishant-cyro Could you please rebase & squash? And sure, I will rebuild before merge

**Comment by nishant-cyro** [comment]
@Mafi88 - done.

**Comment by priyank-gupta** [comment]
:+1:

**Pull request merged**